### ISSUE TITLE
Training LSTM RNN  model after restoration starting again with high loss

### ISSUE THREAD

**Issue by amadupu** [tensorflow/tensorflow#10816]
I have an LSTM based RNN language model where in after the initial training for 1000 iterations, the model is saved as follows
saver = tf.train.Saver()
saver.save(sess,'rnn_model.ckpt',global_step=1000)

I've restored the model by

            saver = tf.train.import_meta_graph("rnn_model.ckpt-1000.meta")
            saver.restore(self.sess, tf.train.latest_checkpoint('./'))
            graph = tf.get_default_graph()
            # restore the operations and placeholder as required from the graph
            # perform retraining

so far everything looks good, the model is restored, but however, when I plot the loss summary while training with the same input data as previous training, the loss again starts with high value as if it is training from start again freshly. I've tried to search all means to find the relevant forums but could not find a proper solution. please advise on the corrective steps

 (related links
https://github.com/tensorflow/tensorflow/issues/6683
https://github.com/fchollet/keras/issues/4875
https://stackoverflow.com/questions/41328769/tensorflow-loss-resets-after-successfully-restored-checkpoint

)

**Comment by amadupu** [comment]
I found the root cause. my input data vectors are modeled differently across each execution due to some issue in the data modeling part. Otherwise, no issue in tensor flow graph after restoration found. After fixing the input data model, I could able to retrain from the last checkpoint properly. I suggest cross checking the input data model for its consistency across multiple executions

**Comment by pickou** [comment]
I ran into the same problem, I have no idea how to solve it. Can you explain your solution in detail? Thank you.

**Comment by vaniakosmos** [comment]
I also had this problem and solved it by specifying random seeds:
```python
np.random.seed(42)
tf.set_random_seed(42)
```
<img width="691" alt="screen shot 2018-01-07 at 22 57 55" src="https://user-images.githubusercontent.com/16745500/34654027-567b3a2c-f3fe-11e7-8b10-ec1b7feae479.png">

2000/3000/3500 iterations are new sessions from restored model. On 3000 I set up seed and on 3500 I kept it and problem gone.

**Comment by riogho** [comment]
@amadupu I am having the same issue as you, any tips on how to solve it? Does it have to do with how I am building the tfrecords from the png? Or is it after that

**Comment by DSRYhh** [comment]
I think [this link](https://stackoverflow.com/a/43670684/5634636) might be helpful.
